File: shareMAC/DynamicProgramming/recursion/maxOfArray.py
def maximum (a, n, max):
	if (n < -1):
		return max
	else:
		if (a[n] > max):
			max = a[n]
		max = maximum(a, n - 1, max)
		return max
print (maximum([100, 11, 6, 9, 3], 4, 0))

File: shareMAC/pepCoding/Recursion + Backtracking/Level-1/maxOfArray.java
import java.util.*;
class findMax
{
	public static void main(String[] args)
	{
		Scanner sc = new Scanner(System.in);
		int n = sc.nextInt();
		int a[] = new int[n];
		for (int i = 0; i < n; i++)
		{
			a[i] = sc.nextInt();
		}
		System.out.println("Maximum is = " + findMax(a, n - 1));
	}
	static int findMax(int a[], int n)
	{
		if (n == 0)
			return a[n];
		else
		{
			if (a[n] >= findMax(a, n - 1))
				return a[n];
			else
				return findMax(a, n - 1);
		}
	}
}
